torchgen/api/autograd.py

```diff
@@ -316,53 +316,53 @@ def is_reference_for_foreach(
     f: NativeFunction,
     function_schema: FunctionSchema,
 ) -> bool:
-    return (
-        f.func.name.name.base.split("_foreach_")[-1] == function_schema.name.name.base
-        and not function_schema.name.name.inplace
-        and all(
-            ref_arg.type in (arg.type, getattr(arg.type, "elem", None))
-            for arg, ref_arg in zip(
-                f.func.arguments.flat_non_out,
-                function_schema.arguments.flat_non_out,
-            )
+    return f.func.name.name.base.split("_foreach_")[
+        -1
+    ] == function_schema.name.name.base and all(
+        ref_arg.type in (arg.type, getattr(arg.type, "elem", None))
+        for arg, ref_arg in zip(
+            f.func.arguments.flat_non_out,
+            function_schema.arguments.flat_non_out,
         )
     )
 
 
 # TODO(crcrpar): Avoid hard coding "Default" ideally.
 def gen_foreach_derivativeinfo(
     foreach_function: NativeFunction,
-    differentiability_infos: Dict[FunctionSchema, Dict[str, DifferentiabilityInfo]],
     functional_info_by_signature: Dict[
         FunctionSchema, Dict[str, DifferentiabilityInfo]
     ],
+    non_functional_info_by_signature: Dict[
```

Comment:
ah this is technically breaking backward compatibility but this function from my perspective isn't general enough for wide adoption.

Would you be okay with suppressing the linter https://github.com/pytorch/test-infra/wiki/BC-Linter#suppression?

Reply:
suppressing sounds ok!